prov/mrail/src/mrail.h

```diff
@@ -128,16 +128,41 @@ mrail_match_recv_handle_unexp(struct mrail_recv_queue *recv_queue, uint64_t tag,
 			      uint64_t addr, char *data, size_t len, void *context);
 
 /* mrail protocol */
-#define MRAIL_HDR_VERSION 1
+#define MRAIL_HDR_VERSION 2
+
+#define MRAIL_PROTO_EAGER	0
+#define MRAIL_PROTO_RNDV	1
+
+#define MRAIL_RNDV_REQ		0
+#define MRAIL_RNDV_ACK		1
 
 struct mrail_hdr {
 	uint8_t		version;
 	uint8_t		op;
-	uint8_t		padding[2];
+	uint8_t		protocol;
+	uint8_t		protocol_cmd;
 	uint32_t	seq;
 	uint64_t 	tag;
 };
 
+#define MRAIL_IOV_LIMIT		5
+
+/* bit 60~63 are provider defined */
+#define MRAIL_RNDV_FLAG		(1ULL << 60)
+
+struct mrail_rndv_hdr {
+	uint64_t		context;
+};
+
+struct mrail_rndv_req {
+	size_t			len;
+	size_t			count;
+	size_t			mr_count;
+	struct fi_rma_iov	rma_iov[MRAIL_IOV_LIMIT];
```

Comment:
Does the value of MRAIL_IOV_LIMIT restrict the number of rails that can be used?  (I'm okay with that, just checking.)

Reply:
No. It only affects the buffer layout. Payload is not divided based on iov count.